### PR TITLE
feat: Responsive Spacing control - Added default Linked support only if connected flag is set to true from setting configs.

### PR DESCRIPTION
**Pull request by RajkiranBagal**
### Description
 - Trello task - https://trello.com/c/or2ndfp5/1029-by-default-link-icon-should-be-disconnected

### Types of changes
 New feature (non-breaking change which adds functionality)

### How has this been tested?
 - Check all ast-responsive-spacing settings if they are linked by default.

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] I've included any necessary tests 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js [hunk 1]
```diff
@@ -85,17 +85,21 @@ const ResponsiveSpacingComponent = props => {
 			id,
 			choices,
 			inputAttrs,
-			unit_choices
+			unit_choices,
+			connected
 		} = props.control.params;
 
+		let connected_class = ( connected ) ? 'connected' : '';
+		let disconnected_class = ( connected ) ? 'disconnected' : '';
```
Comment (premanshup) [inline inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js:93]: @RajkiranBagal Can we combine these two?

Something like this?

`let is_connected_class = ( connected ) ? 'connected' : 'disconnected';`

Comment (RajkiranBagal) [reply inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js]: No these variables need to be used in two different places.

**Review by vrundakansara — APPROVED** [review]

**Review by premanshup — APPROVED** [review]

File: inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js [hunk 2]
```diff
@@ -85,17 +85,21 @@ const ResponsiveSpacingComponent = props => {
 			id,
 			choices,
 			inputAttrs,
-			unit_choices
+			unit_choices,
+			connected
 		} = props.control.params;
 
+		let connected_class = ( false === connected ) ? '' : 'connected';
```
Comment (Nikschavan) [inline inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js:92]: use `camelCase` for variable names for the new variables defined in the PR.

**Review by Nikschavan — CHANGES_REQUESTED** [review]
@RajkiranBagal - Can you also check the conflicts in the PR?

File: inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js (hunk 2 above)
Comment (RajkiranBagal) [reply inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js]: Done

File: inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js [hunk 3]
```diff
@@ -85,17 +85,21 @@ const ResponsiveSpacingComponent = props => {
...
+
 		let itemLinkDesc = __('Link Values Together', 'astra');
 
 		let linkHtml = null;
 		let htmlChoices = null;
 		let respHtml = null;
 
 		if (linked_choices) {
-			linkHtml = <li key={'connect-disconnect' + device} className="ast-spacing-input-item-link disconnected">
+			linkHtml = <li key={'connect-disconnect' + device} className={ `ast-spacing-input-item-link ${disconnected_class}` }>
```
Comment (Nikschavan) [inline inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js:102]: The variable will need to be updated here and below as well

Comment (RajkiranBagal) [reply inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js:102]: Yes, Sorry missed it.

Comment (RajkiranBagal) [reply inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js]: Done

File: inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js [hunk 4]
```diff
@@ -85,17 +85,21 @@ const ResponsiveSpacingComponent = props => {
...
 			choices,
 			inputAttrs,
-			unit_choices
+			unit_choices,
+			connected
 		} = props.control.params;
 
+		let connectedClass = ( false === connected ) ? '' : 'connected';
+		let disconnectedClass = ( false === connected ) ? '' : 'disconnected';
+
```
Comment (Nikschavan) [inline inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js:94]:
```suggestion
```

No need of empty line here as next line is variable declaration.

File: inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js [hunk 5]
```diff
@@ -85,17 +85,21 @@ const ResponsiveSpacingComponent = props => {
...
 		let itemLinkDesc = __('Link Values Together', 'astra');
```
Comment (Nikschavan) [inline inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js:96]:
```suggestion
```

This line can also be removed.

File: inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js (hunk 4 above)
Comment (RajkiranBagal) [reply inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js]: Done

File: inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js (hunk 5 above)
Comment (RajkiranBagal) [reply inc/customizer/extend-custom-controls/src/responsive-spacing/responsive-spacing-component.js]: Done

**Review by Nikschavan — APPROVED** [review]

**Pull request merged**